src/integer/mat_poly_over_z/concat.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright © 2023 Marvin Beckmann
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementations to concatenate two [`MatPolyOverZ`].
+
+use super::MatPolyOverZ;
+use crate::{
+    error::MathError,
+    traits::{Concatenate, GetNumColumns, GetNumRows},
+};
+use flint_sys::fmpz_poly_mat::{fmpz_poly_mat_concat_horizontal, fmpz_poly_mat_concat_vertical};
+
+impl Concatenate for &MatPolyOverZ {
+    type Output = MatPolyOverZ;
+
+    /// Concatenates `self` with `other` vertically, i.e. `other` is added below.
+    ///
+    /// Parameters:
+    /// - `other`: the other matrix to concatenate with `self`
+    ///
+    /// Returns a horizontal concatenation of the two matrices or a
```

Comment:
this probably slipped through the last review as well, can you fix it in MatZ?
```suggestion
    /// Returns a vertical concatenation of the two matrices or a
```